commands/operator-sdk/cmd/add/api.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package add
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-sdk/commands/operator-sdk/cmd/cmdutil"
+	"github.com/operator-framework/operator-sdk/pkg/scaffold"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	apiVersion string
+	kind       string
+)
+
+func NewApiCmd() *cobra.Command {
+	apiCmd := &cobra.Command{
+		Use:   "api",
+		Short: "Adds a new api definition under pkg/apis",
+		Long: `operator-sdk add api --kind=<kind> --api-version=<group/version> creates the
+api definition for a new custom resource under pkg/apis. This command must be run from the project root directory.
```

Comment:
You're right currently it will just overwrite. I'll change to error out if pkg/apis/group/version already exists.

Reply:
Fixed. The command will error out if the path already exists.
```
$ operator-sdk add api --api-version=app.example.com/v1alpha1 --kind=AppService
2018/09/17 14:50:48 /Users/haseeb/work/go-space/src/github.com/example-inc/app-operator/pkg/apis/app/v1alpha1 already exists
```

Reply:
Another case, if I call add api with same `--api-version` but different `--kind`. How does sdk handle this?
What's the correct behavior?

Reply:
Also it seems the code can be messy if we want sdk to add different kind to the same apiVersion because it messes with register.go where we need to add the new types to the existing one.
```go
// addKnownTypes adds the set of types defined in this package to the supplied scheme.
func addKnownTypes(scheme *runtime.Scheme) error {
	scheme.AddKnownTypes(SchemeGroupVersion,
		&App{},
		&AppList{},
	)
	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
	return nil
}
```
https://github.com/hasbro17/operator-sdk-samples/blob/refactor/app-operator-using-controller-runtime/app-operator/pkg/apis/app/v1alpha1/register.go#L24-L27

Reply:
Very Nice. Let's add that in a new PR. Also we probably need to document what cases can happen and what's the sdk's behavior in the `sdk add api` command descriptions.